yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala

```diff
@@ -87,6 +92,21 @@ private[spark] class ClientArguments(args: Array[String], sparkConf: SparkConf)
       throw new IllegalArgumentException(
         "You must specify at least 1 executor!\n" + getUsageMessage())
     }
+    if (isClusterMode) {
+      for (key <- Seq(amMemKey, amMemOverheadKey)) {
+        if (sparkConf.getOption(key).isDefined) {
+          println(s"$key is set but does not apply in cluster mode.")
```

Comment:
I don't think you want to use `println` here. Also, I'm always a little conflicted about these logs, since they'll show up when you set these values in the default conf file (which would apply to a lot of different jobs unless they explicitly override the conf file). But not a big deal.